pkg/rancher-desktop/config/settingsImpl.ts

```diff
@@ -447,32 +468,47 @@ export const updateTable: Record<number, (settings: any) => void> = {
   },
 };
 
-function migrateSettingsToCurrentVersion(settings: Settings) {
+function migrateSettingsToCurrentVersion(settings: Record<string, any>): Settings {
   if (Object.keys(settings).length === 0) {
     return defaultSettings;
   }
-  let loadedVersion = settings.version || 0;
+  settings = migrateSpecifiedSettingsToCurrentVersion(settings) as Settings;
 
-  if (loadedVersion < CURRENT_SETTINGS_VERSION) {
-    for (; loadedVersion < CURRENT_SETTINGS_VERSION; loadedVersion++) {
-      if (updateTable[loadedVersion]) {
-        updateTable[loadedVersion](settings);
-      }
+  return _.defaultsDeep(settings, defaultSettings);
+}
+
+/**
+ * Used to migrate a settings payload from an earlier version to the current one.
+ * Input payloads are expected to come from either the argument to `rdctl api settings -X PUT ...`
+ * or a deployment profile.
+ *
+ * The contents of settings files go through the unexported function `migrateSettingsToCurrentVersion`
+ * which assigns any missing defaults at the end. This function does not fill in missing values.
```

Comment:
This should be phrased in a way that doesn't rely on knowledge of implementation details (in this case, that `migrateSettingsToCurrentVersion` is a thing.  We probably want something closer to:

> Migrate a settings payload from an earlier version to the current one.  Both the input as well the result may only contain partial settings; this is not suitable for migrating the settings file.

Reply:
But `migrateSettingsToCurrentVersion` is the immediately preceding function. Some documentation is needed to explain the difference between the two.

Reply:
Right, but when you're looking at the docs from a tooltip (hovering over a call site) the fact that the other function happens to be immediately preceding is irrelevant.